### PR TITLE
add e2e dashboard folder

### PR DESCRIPTION
**Pull request by Aman-Codes**
# Description
Added the new e2e dashboard source code and build files which were earlier at [Aman-Codes/e2e-dashboard](https://github.com/Aman-Codes/e2e-dashboard).

# Screenshots
image
image
image

# Hosted URL
- https://aman-codes.github.io/litmus-e2e/dashboard/build/
- https://aman-codes.github.io/e2e-dashboard/

### CODE REVIEW

**Comment by Aman-Codes** [comment]
> Thanks @Aman-Codes for the PR. Can you share the link of GitHub Pages hosted from your [forked e2e repository](https://github.com/Aman-Codes/litmus-e2e/tree/gh-pages)?

Hi @uditgaurav https://aman-codes.github.io/litmus-e2e contains the current dashboard and https://aman-codes.github.io/litmus-e2e/dashboard/build/ contains the new e2e dashboard

**Comment by uditgaurav** [comment]
Can we update the pipeline description with right pipeline names like:
## For:

image
it can be "( GO BDDs) for the node-level pipelines"

File: dashboard/build/404.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      // If you're creating a Project Pages site and NOT using a custom domain,
+      // then set pathSegmentsToKeep to 1 (enterprise users may need to set it to > 1).
+      // This way the code will only replace the route part of the path, and not
+      // the real directory in which the app resides, for example:
+      // https://username.github.io/repo-name/one/two?a=b&c=d#qwe becomes
+      // https://username.github.io/repo-name/?/one/two&a=b~and~c=d#qwe
+      // Otherwise, leave pathSegmentsToKeep as 0.
+      var pathSegmentsToKeep = 1;
+
+      var l = window.location;
```
Comment (oumkale) [inline dashboard/build/404.html:27]: You may use some relevant variables

File: dashboard/build/404.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      var p = l.protocol + '//' + l.hostname + (l.port ? ':' + l.port : '') +
```
Comment (oumkale) [inline dashboard/build/404.html:28]: Update relevant variable name for all

File: dashboard/build/index.css [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  border: 8px solid #d9d9d9;
+  border-radius: 50%;
+  border-top: 8px solid #5b44ba;
+  width: 60px;
+  height: 60px;
+  -webkit-animation: spin 2s linear infinite; /* Safari */
+  animation: spin 2s linear infinite;
+}
+
+/* Safari */
```
Comment (oumkale) [inline dashboard/build/index.css:26]: Will you remove unwated comments

Comment (oumkale) [inline dashboard/build/index.css:26]: Add relevant info about CSS styles Like where it is used

**Comment by Jonsy13** [comment]
Can we remove the build folder, we can use the github actions to build and push the same to gh-pages branch instead of pushing it manually, that way we will be able to test the build as well?

File: dashboard/build/service-worker.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+self.addEventListener('message', (event) => {
+  if (event.data && event.data.type === 'SKIP_WAITING') {
+    self.skipWaiting();
+  }
+});
+
+workbox.core.clientsClaim();
+
+/**
```
Comment (oumkale) [inline dashboard/build/service-worker.js:28]: Will you remove unwanted comments and add relevant comments

File: dashboard/package.json [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    "@material-ui/icons": "^4.9.1",
+    "@testing-library/jest-dom": "^5.11.10",
+    "@testing-library/react": "^11.2.6",
+    "axios": "^0.21.4",
+    "date-fns": "^2.24.0",
+    "i18next": "^19.7.0",
+    "i18next-browser-languagedetector": "^6.0.1",
+    "i18next-http-backend": "^1.0.19",
+    "js-yaml": "^4.1.0",
+    "litmus-ui": "1.6.0",
```
Comment (Jonsy13) [inline dashboard/package.json:18]: You may also update the litmus-ui package, will enhance the performance.

**Comment by oumkale** [comment]
@Aman-Codes There are a couple of files and CSS styling are reusable cc: @Jonsy13

File: dashboard/public/locales/en/translation.yaml [hunk 6]
```diff
@@ -0,0 +1,1463 @@
+getStarted:
```
Comment (Jonsy13) [inline dashboard/public/locales/en/translation.yaml:1]: This file is not required, we are not using translation in dashboards, right? Also this file is for Portal not for dahsboard.

File: dashboard/src/api/filterWorkflow.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { readableNameConverter } from "shared/helper";
+
+export const filterWorkflow = (responseData) => {
+  if(responseData) {
+    const total_count = responseData.total_count;
+    const scheduledData = [];
+    const manualData = [];
+    responseData.workflows.forEach((element) => {
+      if (element.name.match(/^Scheduled.*Pipeline$/) != null) {
```
Comment (Jonsy13) [inline dashboard/src/api/filterWorkflow.js:9]: It will be Nightly? @uditgaurav

File: dashboard/src/api/sendRequest.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+
+export const sendGetRequest = (url) => {
+  console.log("url inside sendRequest is", url);
```
Comment (Jonsy13) [inline dashboard/src/api/sendRequest.js:4]: Please remove all console logs

File: dashboard/src/api/sendRequest.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    headers: {
+      Accept: 'application/vnd.github.v3+json'
+    },
+    // auth: {
+    //   username: process.env.REACT_APP_GITHUB_USERNAME,
+    //   password: process.env.REACT_APP_GITHUB_PAT
+    // }
+  })
+  .then((response) => {
+    console.log("response inside sendRequest is", response.data);
```
Comment (Jonsy13) [inline dashboard/src/api/sendRequest.js:17]: Here also

File: dashboard/src/api/sendRequest.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    return response.data;
+  })
+  .catch((error) => {
+    console.log("Error inside sendRequest is", error);
```
Comment (Jonsy13) [inline dashboard/src/api/sendRequest.js:21]: Only display the error in this case... also console.error() should be used

File: dashboard/src/components/CustomCard/index.jsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  pos: {
+    marginBottom: 12,
+    color: '#0000008a'
+  },
+});
+
+const CustomCard = ({ data, category }) => {
+  const classes = useStyles();
+  console.log("data inside CustomCard", data);
```
Comment (Jonsy13) [inline dashboard/src/components/CustomCard/index.jsx:31]: Same here

File: dashboard/src/components/Header/ProfileDropDown.jsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
```
Comment (Jonsy13) [inline dashboard/src/components/Header/ProfileDropDown.jsx:1]: These files are not required

File: dashboard/src/components/Header/ProjectDropDown.jsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { IconButton, Popover, Typography } from '@material-ui/core';
```
Comment (Jonsy13) [inline dashboard/src/components/Header/ProjectDropDown.jsx:2]: Not required

File: dashboard/src/components/Header/ProjectDropDownItems.jsx [hunk 14]
```diff
@@ -0,0 +1,208 @@
+import { useQuery } from '@apollo/client';
+import {
```
Comment (Jonsy13) [inline dashboard/src/components/Header/ProjectDropDownItems.jsx:2]: Not required

File: dashboard/src/components/Header/index.jsx [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            </div>
+            <div className={classes.slackIcon}>
+              <a href="https://slack.litmuschaos.io/" target="_blank" rel="noopener noreferrer">
+                <img src={SlackLogo} alt="slack logo" />
+              </a>
+            </div>
+            <div>
+              <a href="https://github.com/litmuschaos/litmus" target="_blank" rel="noopener noreferrer">
+                litmuschaos/litmus <br/>
+                <LocalOfferIcon style={{paddingTop: 4}} /> 2.1.1 <StarBorder style={{paddingTop: 4}} /> 2.2k <AccountTree style={{paddingTop: 4}} /> 416
```
Comment (Jonsy13) [inline dashboard/src/components/Header/index.jsx:69]: Version is not dynamic

File: dashboard/src/components/Header/styles.js [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+  // Logo
+  nounderline: {
+    textDecoration: 'none',
+  },
+  chaosText: {
+    fontSize: '1.625rem',
+    color: theme.palette.text.secondary,
+    fontWeight: 600,
+  },
+  projectDropdown: {
```
Comment (Jonsy13) [inline dashboard/src/components/Header/styles.js:36]: Styles not required

File: dashboard/src/components/Menu/index.jsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
```
Comment (Jonsy13) [inline dashboard/src/components/Menu/index.jsx:1]: This is also not required.

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

**Comment by Jonsy13** [comment]
@Aman-Codes Please remove everything which is not related to dashboard

**Comment by Jonsy13** [comment]
Screenshot from 2021-10-13 11-18-56

For Portal pipeline, the commit ids should be taken from litmuschaos/litmus Repo

File: dashboard/src/components/Header/ProjectDropDownItems.jsx [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+  label,
+  secondaryLabel,
+  selected,
+}) => {
+  const classes = useStyles();
+  const theme = useTheme();
+  const [copying, setCopying] = useState<boolean>(false);
+
+  function fallbackCopyTextToClipboard(text) {
+    // eslint-disable-next-line no-alert
```
Comment (oumkale) [inline dashboard/src/components/Header/ProjectDropDownItems.jsx:33]: Same

File: dashboard/src/components/Header/index.jsx [hunk 19]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Typography } from '@material-ui/core';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import LocalOfferIcon from '@material-ui/icons/LocalOffer';
+import StarBorder from '@material-ui/icons/StarBorder';
+import AccountTree from '@material-ui/icons/AccountTree';
+// import ProfileDropdown from './ProfileDropDown';
+// import ProjectDropdown from './ProjectDropDown';
```
Comment (oumkale) [inline dashboard/src/components/Header/index.jsx:10]: Same

File: dashboard/src/components/Header/index.jsx [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+            <div>
+              <a href="https://github.com/litmuschaos/litmus" target="_blank" rel="noopener noreferrer">
+                litmuschaos/litmus <br/>
+                <LocalOfferIcon style={{paddingTop: 4}} /> 2.1.1 <StarBorder style={{paddingTop: 4}} /> 2.2k <AccountTree style={{paddingTop: 4}} /> 416       
+              </a>              
+            </div>
+          </div>
+          {/* <div className={classes.details}>
+            <ProjectDropdown />
+            <ProfileDropdown />
```
Comment (oumkale) [inline dashboard/src/components/Header/index.jsx:75]: Remove this if it is not required

File: dashboard/src/components/SideBar/index.jsx [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+      if(prevValue === value) {
+        return "none";
+      }
+      return value;
+    });
+  }
+  // const handleScheduled = () => {
+  //   setOpenScheduled((prevState) => !prevState);
+  // }
+  // const handleManual = () => {
```
Comment (oumkale) [inline dashboard/src/components/SideBar/index.jsx:30]: Same

File: dashboard/src/components/SideBar/index.jsx [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+          key="litmusDocs"
+          handleClick={() => {
+            window.open('https://docs.litmuschaos.io/');
+          }}
+          label="Litmus Docs"
+        >
+          <DocsIcon />
+        </CustomisedListItem>
+        {/* <CustomisedListItem
+          key="litmusAPIDocs"
```
Comment (oumkale) [inline dashboard/src/components/SideBar/index.jsx:121]: Remove all if it is not required

File: dashboard/src/components/Table/TimelineComponent.jsx [hunk 23]
```diff
@@ -0,0 +1,123 @@
...
+  running: '#3a97d4',
+  pending: '#949ab7'
+}
+
+const TimelineItemComponent = ({ step, connectorLine=true }) => {
+  const classes = useStyles();
+  /*
+    Possible status values are:
+    completed
+    in_progress
```
Comment (oumkale) [inline dashboard/src/components/Table/TimelineComponent.jsx:26]: Remove all if it is not required

File: dashboard/src/components/Table/helper.js [hunk 24]
```diff
@@ -0,0 +1,44 @@
+/*
+  Possible conclusion values are:
+  action_required
+  cancelled
+  failure
+  neutral
```
Comment (oumkale) [inline dashboard/src/components/Table/helper.js:6]: Same

File: dashboard/src/containers/app/Routes.jsx [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+const Routes = () => {
+  return (
+    <Scaffold>
+      <SuspenseLoader style={{ height: '80vh' }}>
+        <Switch>
+          <Route exact path="/" render={(props) => <HomePage {...props} />} />
+          <Route exact path="/manual-runs" render={(props) => <ManualRuns {...props} />} />
+          {/* <Route exact path="/manual-runs/:pipelineName" render={(props) => <ManualRuns {...props} />}/> */}
+          <Route exact path="/scheduled-runs" render={(props) => <ScheduledRuns {...props} />} />
+          {/* <Route exact path="/scheduled-runs/:pipelineName" render={(props) => <ScheduledRuns {...props} />} /> */}
```
Comment (oumkale) [inline dashboard/src/containers/app/Routes.jsx:20]: Remove unwanted routes

File: dashboard/src/containers/layouts/Scaffold/index.jsx [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
+      <CssBaseline />
+      <header className={classes.header}>
+        <Header />
+      </header>
+      {/* <aside className={classes.sidebar}>
+        <SideBar scheduledData={scheduledData} manualData={manualData}/>
```
Comment (oumkale) [inline dashboard/src/containers/layouts/Scaffold/index.jsx:17]: Remove this if not require

**Comment by oumkale** [comment]
Hi @Aman-Codes,

 - Add function details for every function used
 - Remove all unwanted comments and unused code
 - Follow proper lint format, Many places spaces, newline and all not formatted
 - Use all heading and info from the manifest. Take ref from litmus-portal
 - Make command functions reusable, same for CSS. cc: @Jonsy13

File: dashboard/src/components/CustomCard/index.jsx [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+          <PlayCircleFilled style={{ marginBottom: '-0.3rem'}} /> {data?.readableName}
+        </Typography>
+        <img src={data?.badge_url} alt="status of pipeline" />
+      </CardContent>
+      <CardActions>
+        <Link to={{
+            pathname: `/${category}`,
+            state: { id: data?.id,  readableName: data?.readableName}
+        }}>
+          <Button size="small">Pipeline Details</Button>
```
[the remark this answers is not in the document]
Comment (Aman-Codes) [reply dashboard/src/components/CustomCard/index.jsx:46]: Did you meant to put all the text visible on user screen under a single file like https://github.com/litmuschaos/litmus/blob/master/litmus-portal/frontend/public/locales/en/translation.yaml ?

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

File: dashboard/public/404.html [hunk 28]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Single Page Apps for GitHub Pages</title>
+    <script type="text/javascript">
+      var pathSegmentsToKeep = 1;
+
+      var location = window.location;
+      var updatedLocation = l.protocol + '//' + l.hostname + (l.port ? ':' + l.port : '') +
```
[the remark this answers is not in the document]
Comment (Aman-Codes) [reply dashboard/public/404.html]: Updated It

**Review by Jonsy13 — APPROVED** [review]
Please check the build process. @uditgaurav @Aman-Codes 
Pushing the build folder is not a good practice. We cannot review the autogenerated build files.

**Pull request merged**